cadl-tests/src/main/java/com/cadl/builtin/BuiltinClient.java

```diff
@@ -172,21 +181,36 @@ public Response<Void> writeWithResponse(BinaryData body, RequestOptions requestO
      */
     @Generated
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Builtin read(OffsetDateTime dateTime, String filter) {
+    public Builtin read(
+            String queryParam,
+            String queryParamEncoded,
+            OffsetDateTime dateTime,
+            String filter,
+            String queryParamOptional,
+            String queryParamOptionalEncoded) {
         // Generated convenience method for readWithResponse
         RequestOptions requestOptions = new RequestOptions();
         if (dateTime != null) {
             requestOptions.setHeader("x-ms-date", String.valueOf(new DateTimeRfc1123(dateTime)));
         }
         if (filter != null) {
-            requestOptions.addQueryParam("filter", filter);
+            requestOptions.addQueryParam("filter", filter, false);
         }
-        return readWithResponse(requestOptions).getValue().toObject(Builtin.class);
+        if (queryParamOptional != null) {
+            requestOptions.addQueryParam("query-opt", queryParamOptional, false);
+        }
+        if (queryParamOptionalEncoded != null) {
+            requestOptions.addQueryParam("query-opt-encoded", queryParamOptionalEncoded, true);
+        }
```

Comment:
Here is the Cadl logic in convenience API.

If query parameter is already URL encoded, use the 3rd parameter to set `encoded` to `true` on `RequestOptions.addQueryParam`.